api4/file_test.go

```diff
@@ -609,10 +621,16 @@ func TestUploadFiles(t *testing.T) {
 							require.Nil(t, err)
 							if !bytes.Equal(data, expected) {
 								tf, err := ioutil.TempFile("", fmt.Sprintf("test_%v_*_%s", i, name))
+								defer tf.Close()
+								require.Nil(t, err)
+								wBytes, err := io.Copy(tf, bytes.NewReader(data))
 								require.Nil(t, err)
-								_, _ = io.Copy(tf, bytes.NewReader(data))
-								tf.Close()
-								t.Errorf("Actual data mismatched %s, written to %q", name, tf.Name())
+								require.Equal(t, int64(len(data)), wBytes)
```

Comment:
A problem with this line is that if the lengths are unequal, we would never know what the image differences are. And since, we are already logging the lengths in the subsequent lines, I think that information is anyways present in the logs. I would suggest to just remove this.